upf/flowtable.h

```diff
@@ -88,7 +88,11 @@ typedef struct
 typedef struct
 {
   u32 pkts;
+  u32 pkts_unreported;
   u64 bytes;
+  u64 bytes_unreported;
+  u64 l4_bytes;
+  u64 l4_bytes_unreported;
```

Comment:
I really can't see why `l4_bytes` and `l4_bytes_unreported` is needed. It seems that nothing is using `l4_bytes` and both fields will have the same value.
THe same might apply to pkts and bytes.

Reply:
That actually looks like a bug b/c at some point, `l4_bytes_unreported` were being cleared for the flow when the report was being sent, but somehow I lost that part and somehow the test did not detect the issue. I will recheck the code, thanks. Plain `l4_bytes` is for the "total" counter starting from the beginning of the flow, presently the fields that are using it aren't added to the existing templates but we might easily need them.

Reply:
Rechecked it and there's no bug except that the tests needed a bit of adjustment to check it's all working correctly (will push the changes).
Note that for `initiatorPackets` / `responderPackets` / `initiatorOctets` / `responderOctets` we use the macro `IPFIX_VALUE_DELTA_U64`
https://github.com/travelping/upg-vpp/blob/feature/ipfix/upf/upf_ipfix_templates.h#L127
Which does reset the corresponding flow field to zero after copying the value to the IPFIX report
https://github.com/travelping/upg-vpp/blob/feature/ipfix/upf/upf_ipfix_templates.h#L68
Thus, `..._unreported` here stands for the delta fields for which the values are reset to zero during each IPFIX report (that is, "unreported" in the sense "the amount which is not reported yet")

Reply:
Sure, that applies to l4_bytes_unreported and pkts_unreported. There is no user for l4_bytes and bytes_unreported. I wonder if that is an oversight?